src/Integration.Vsix.UnitTests/SonarLintTagger/AccumulatingIssueConsumerTests.cs

```diff
@@ -0,0 +1,182 @@
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.VisualStudio.Text;
+using Moq;
+using SonarLint.VisualStudio.Core.Analysis;
+using SonarLint.VisualStudio.Integration.Vsix;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.SonarLintTagger
+{
+    [TestClass]
+    public class AccumulatingIssueConsumerTests
+    {
+        private static readonly IAnalysisIssue ValidIssue = CreateIssue(startLine: 1, endline: 1);
+        private static readonly ITextSnapshot ValidTextSnapshot = CreateSnapshot(lineCount: 10);
+        private const string ValidFilePath = "c:\\myfile.txt";
+
+        [TestMethod]
+        public void Ctor_InvalidArgs_Throws()
+        {
+            AccumulatingIssueConsumer.OnIssuesChanged validCallback = _ => { };
+
+            Action act = () => new AccumulatingIssueConsumer(null, ValidFilePath, validCallback);
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("analysisSnapshot");
+
+            act = () => new AccumulatingIssueConsumer(ValidTextSnapshot, null, validCallback);
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("analysisFilePath");
+
+            act = () => new AccumulatingIssueConsumer(ValidTextSnapshot, ValidFilePath, null);
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("onIssuesChangedCallback");
+        }
+
+        [TestMethod]
+        public void Accept_WrongFile_CallbackIsNotCalled()
+        {
+            var callbackSpy = new OnIssuesChangedCallbackSpy();
+            var issues = new IAnalysisIssue[] { ValidIssue };
+
+            var testSubject = new AccumulatingIssueConsumer(ValidTextSnapshot, "c:\\file1.txt", callbackSpy.Callback);
+
+            using (new AssertIgnoreScope())
+            {
+                testSubject.Accept("wrong file", issues);
+            }
+
+            callbackSpy.CallCount.Should().Be(0);
+        }
+
+        [TestMethod]
+        [DataRow(-1, 10, false)] // start line < 1
```

Comment:
I think it's worth giving a different number than 10 for the last line, just so it would be more readable in terms of zero-based/one-based. (-1, 1) for example.

Reply:
I don't understand, but ok.